capellambse_context_diagrams/collectors/class_tree.py

```diff
@@ -0,0 +1,88 @@
+# SPDX-FileCopyrightText: 2022 Copyright DB Netz AG and the capellambse-context-diagrams contributors
+# SPDX-License-Identifier: Apache-2.0
+"""This submodule defines the collector for the Class-Tree diagram."""
+from __future__ import annotations
+
+import collections.abc as cabc
+import typing as t
+
+from capellambse import helpers
+from capellambse.model.crosslayer import information
+
+from .. import _elkjs, context
+from . import generic, makers
+
+LAYOUT_OPTIONS: _elkjs.LayoutOptions = {
+    "algorithm": "layered",
+    "edgeRouting": "ORTHOGONAL",
+    "elk.direction": "DOWN",
+    "partitioning.activate": True,
+    "edgeLabels.sideSelection": "ALWAYS_DOWN",
+}
+
+
+def collector(
+    diagram: context.ContextDiagram, params: dict[str, t.Any] | None = None
+) -> _elkjs.ELKInputData:
+    """Return the class tree data for ELK."""
+    params = params or {}
+    assert isinstance(diagram.target, information.Class)
+    data = generic.collector(diagram, no_symbol=True)
+    if params.get("partitioning", False):
+        data["children"][0]["layoutOptions"] = {}
+        data["children"][0]["layoutOptions"]["elk.partitioning.partition"] = 0
+    data["layoutOptions"] = LAYOUT_OPTIONS
```

Comment:
Note this will overwrite the global `LAYOUT_OPTIONS` dict with the keys provided, which could potentially cause data races. We should make a copy here.

```suggestion
    data["layoutOptions"] = LAYOUT_OPTIONS.copy()
```